### ISSUE TITLE
goreleaser 1.17.x broken in WSL with Docker Desktop installed

### ISSUE THREAD

**Issue by hhromic** [goreleaser/goreleaser#3945]
### What happened?

Since `1.17.0`, goreleaser does not work anymore under Windows Subsystem for Linux (WSL) with Docker Desktop installed.
The reason is this recent PR in goreleaser:
* #3901

Which introduced the [automaxprocs](https://github.com/uber-go/automaxprocs) library into goreleaser, which does not currently work under WSL + Docker Desktop:
* https://github.com/uber-go/automaxprocs/issues/65

Goreleaser refuses to run because it cannot set GOMAXPROCS. See below in the reproduction case.

The problem should be solved in the automaxprocs library, but in the meantime goreleaser is unusable since 1.17.0 out of the box.

A workaround is to provide an explicit GOMAXPROCS value as an environment variable. For example:
```
$ GOMAXPROCS=0 ./goreleaser  --version
goreleaser version 1.17.0
commit: ffefd6c4ae257d2e9aeccb5786ba66401cc1bd78
built at: 2023-04-10T11:56:39Z
built by: goreleaser
goos: linux
goarch: amd64
module version: v1.17.0, checksum: h1:CcA7EOcljdYhpEY2MUI1je6swCTCAlubxZk1VCbPKBY=

https://goreleaser.com
```

### How can we reproduce this?

To reproduce, just run this into a WSL session (with Docker Desktop also installed in the machine):
```
$ mkdir tmp
$ cd tmp
$ curl -sSL https://github.com/goreleaser/goreleaser/releases/download/v1.17.0/goreleaser_Linux_x86_64.tar.gz | tar zxvf -
$ ./goreleaser --version
  ⨯ failed to set GOMAXPROCS                         error=invalid format for MountPoint: "991 77 0:252 / /Docker/host rw,noatime - 9p drvfs rw,dirsync,aname=drvfs;path=C:\\Program Files\\Docker\\Docker\\resources;symlinkroot=/mnt/,mmap,access=client,msize=262144,trans=virtio"
```

### goreleaser version

```bash
goreleaser version 1.17.0
commit: ffefd6c4ae257d2e9aeccb5786ba66401cc1bd78
built at: 2023-04-10T11:56:39Z
built by: goreleaser
goos: linux
goarch: amd64
module version: v1.17.0, checksum: h1:CcA7EOcljdYhpEY2MUI1je6swCTCAlubxZk1VCbPKBY=

https://goreleaser.com
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caarlos0** [comment]
see https://github.com/goreleaser/goreleaser/pull/3946

**Comment by hhromic** [comment]
As a follow up, the problem is actually in Docker Desktop for Windows. See this [comment](https://github.com/uber-go/automaxprocs/issues/65#issuecomment-1512726580) for more info and a workaround to avoid the warning until they fix it upstream.

**Comment by caarlos0** [comment]
ah cool, either way, warning seems more reasonable than hard failing... 

thanks for investigating 🤘

**Comment by hhromic** [comment]
@caarlos0 yes, definitively goreleaser should emit a warning only and not fail when not being able to automatically configure GOMAXPROCS. The user can always set it manually in such cases.

Thanks for your PR and the recent release!
It works well for me now (despite the warning that should disappear whenever Docker fixes the issue properly).

**Comment by hhromic** [comment]
@caarlos0 the warning is now fixed in [automaxprocs v1.5.3](https://github.com/uber-go/automaxprocs/releases/tag/v1.5.3).
Whenever you get a chance, could you update the dependency in goreleaser? :)
The warning is really annoying in WSL2 😢.
